backend/infrahub/graphql/queries/diff/tree.py

```diff
@@ -409,11 +410,20 @@ async def resolve(
             include_parents=include_parents,
             limit=limit,
             offset=offset,
+            tracking_id=NameTrackingId(name) if name else None,
             include_empty=True,
         )
         if not enriched_diffs:
             return None
-        enriched_diff = enriched_diffs[0]
```

Comment:
we allow users to request a same branch "diff", which is really just a record of all the changes on a given branch over the given time period, but that is not the most common use case. this logic gives preference to diffs across branches so that we will only return same-branch "diff"s if explicitly requested